file/src/test/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSinkSpec.scala

```diff
@@ -0,0 +1,164 @@
+            lastFilename = Some(newName)
+            Some(fs.getPath(newName))
+          } else {
+            None
+          }
+        }
+    }
+    // #LogRotationSink-timebased-sample
+
+    val pathGeneratorFunction = timeBasedRotationFunction
+
+    // #LogRotationSink-sample
+    val completion = Source(Seq("test1", "test2", "test3", "test4", "test5", "test6").toList)
+      .map(ByteString(_))
+      .runWith(LogRotatorSink(pathGeneratorFunction))
+    // #LogRotationSink-sample
+
+  }
+
+}
+
+class LogRotatorSinkSpec
+    extends TestKit(ActorSystem("LogRotatorSinkSpec"))
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll {
+
+  override protected def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
+    fs.close()
+  }
+
+  val settings = ActorMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorMaterializer(settings)
+  val fs = Jimfs.newFileSystem("LogRotatorSinkSpec", Configuration.unix())
+
+  val TestLines = {
+    val b = ListBuffer[String]()
+    b.append("a" * 1000 + "\n")
+    b.append("b" * 1000 + "\n")
+    b.append("c" * 1000 + "\n")
+    b.append("d" * 1000 + "\n")
+    b.append("e" * 1000 + "\n")
+    b.append("f" * 1000 + "\n")
+    b.toList
+  }
+
+  val testByteStrings = TestLines.map(ByteString(_))
+
+  "LogRotatorSink" must {
+    "write lines to a single file" in assertAllStagesStopped {
+      var files = Seq.empty[Path]
+      val testFunction = () => {
+        var fileName: String = null
+        (x: ByteString) =>
+          {
+            if (fileName == null) {
+              val path = Files.createTempFile(fs.getPath("/"), "test", ".log")
+              files :+= path
+              fileName = path.toString
+              Some(path)
+            } else {
+              None
+            }
+          }
+      }
+      val completion = Source(testByteStrings).runWith(LogRotatorSink(testFunction))
+      val result = Await.result(completion, 3.seconds)
+
+      val sizes = readUpFileSizesAndClean(files)
+      sizes should contain theSameElementsAs Seq(6006L)
+    }
+    "write lines to multiple files due to filesize" in assertAllStagesStopped {
+      var files = Seq.empty[Path]
+      val testFunction = () => {
+        val max = 2002
+        var size: Long = max
+        (x: ByteString) =>
+          {
+            if (size + x.size > max) {
+              val path = Files.createTempFile(fs.getPath("/"), "test", ".log")
+              files :+= path
+              size = x.size
+              Some(path)
+            } else {
+              size += x.size
+              None
+            }
+          }
+      }
+      val completion =
+        Source(testByteStrings).runWith(LogRotatorSink(testFunction))
+      val result = Await.result(completion, 3.seconds)
+
+      val sizes = readUpFileSizesAndClean(files)
+      sizes should contain theSameElementsAs Seq(2002L, 2002L, 2002L)
```

Comment:
Would be nice to assert that the file contents is what we expect it to be as well, just to be sure the boundary didn't duplicate data or something like that.